lib/spack/spack/cmd/style.py

```diff
@@ -312,33 +343,49 @@ def run_black(black_cmd, file_list, args):
     # run in chunks of 100 at a time to avoid line length limit
     # filename parameter in config *does not work* for this reliably
     for chunk in grouper(file_list, 100):
-        packed_args = check_fix_args + tuple(chunk)
+        packed_args = black_args + tuple(chunk)
         output = black_cmd(*packed_args, fail_on_error=False, output=str, error=str)
         returncode |= black_cmd.returncode
 
         rewrite_and_print_output(output, args, pat, replacement)
 
     print_tool_result("black", returncode)
+
     return returncode
 
 
 def style(parser, args):
+    # ensure python version is new enough
+    if sys.version_info < (3, 6):
+        tty.die("spack style requires Python 3.6 or later.")
```

Comment:
Where does this restriction come from? I know black has always required Python 3.6+, but can't the other tools be run with older Python?

Reply:
Yep nice catch -- copy-paste error.

Reply:
They can, but I figured it was easier to use an upper bound, as we will soon be requiring black.  Do you want it broken down by python support?

Reply:
Personally, I wouldn't check at all. Just error out if a dependency is missing, and let people try to install whatever they can with their Python version.

Reply:
With bootstrapping, this basically means they'll run `spack style` and it'll try to bootstrap these dependencies on every run instead of just once.  I could optimize that and make it smarter, but for now can we keep this?

Reply:
I'm fine with whatever